### ISSUE TITLE
Reproducing paper

### ISSUE THREAD

**Issue by rolson24** [kevtimova/VJ-VCR#1]
Hi authors,

This is a super interesting paper. A while ago I tried to use the V-JEPA training method to pre-train a VisionMamba SSM on SSv2, but ran into significant issues with representation collapse. I saw that you published this paper and that it did a much better job of preventing representation collapse so I was interested in it. I am trying to reproduce the paper (or at least the MovingMNIST experiments) but I am confused about how you generate the latent vector in the non-deterministic setting. I see that in one mode the latent vector is a one-hot encoding that directly encodes the direction switch, but in the other mode, it is not clear how you select the sparse latent vector during training. I know that during inference you use the FISTA algorithm to optimize the latent vector for each sample, but during training do you do the same thing? Or do you learn the latent vector during training and then use the FISTA algorithm only during inference?

**Comment by kevtimova** [comment]
Hi Raif,

Thank you for reaching out and for your interest in the paper and reproducing the results!

In the case of the non-deterministic MNIST dataset, we infer the sparse latent vectors (using the FISTA algorithm) both during training and evaluation of the VJ-VCR model. So, even during training, for a set of samples, we perform inference with FISTA to find the sparse latents that minimize the objective (prediction loss in the representation space). 

In case it is helpful, please refer to section 2.2 of my earlier work for details on FISTA: https://arxiv.org/pdf/2112.09214. An implementation is available here: https://github.com/kevtimova/deep-sparse/blob/70562a3bc788f99117392b5c41d36224adee4178/utils.py#L141. 

Please let me know if you have any further questions!
